src/views/Home.vue

```diff
@@ -0,0 +1,365 @@
+        <ion-label>{{ $t("Something went wrong while fetching the order details") }}</ion-label>
+      </div>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script lang="ts">
+import {
+  alertController,
+  IonButton,
+  IonCard,
+  IonContent,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonNote,
+  IonPage,
+  IonSelect,
+  IonSelectOption,
+  IonThumbnail,
+  modalController,
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { useRouter } from "vue-router";
+import { mapGetters, useStore } from 'vuex'
+import { OrderService } from "@/services/OrderService";
+import { translate } from "@/i18n";
+import { hasError, showToast } from "@/utils";
+import Image from "@/components/Image.vue";
+import ShipmentAddressModal from "@/views/ShipmentAddressModal.vue";
+import { ProductService } from "@/services/ProductService";
+import PickupLocationModal from "./PickupLocationModal.vue";
+
+export default defineComponent({
+  name: "Home",
+  components: {
+    Image,
+    IonButton,
+    IonCard,
+    IonContent,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonNote,
+    IonSelect,
+    IonSelectOption,
+    IonThumbnail,
+    IonPage,
+  },
+  data() {
+    return {
+      order: {} as any,
+      products: {} as any,
+      productIds: [] as any,
+      deliveryMethods: [
+        {
+          name: 'Store pickup',
+          value: 'STOREPICKUP'
+        },
+        {
+          name: 'Shipping',
+          value: 'STANDARD'
+        }
+      ] as any,
+      areChangesSaved: false
+    }
+  },
+  computed: {
+    ...mapGetters({
+      deliveryMethod: 'user/getDeliveryMethod'
+    })
+  },
+  async mounted() {
+    await this.getOrder();
+  },
+  methods: {
+    async getOrder() {
+      let resp;
+      try {
+        resp = await OrderService.getOrder(this.$route.params.orderId);
+        if (resp.status === 200 && !hasError(resp) && resp.data) {
+          this.order = resp.data;
+          let productIds: any = new Set();
+          this.order.shipGroup.map((group: any) => {
+            group.selectedShipmentMethodTypeId = group.shipmentMethodTypeId;
+            group.items.map((item: any) => {
+              if (item.productId) productIds.add(item.productId);
+            })
+          })
+          this.productIds = [...productIds]
+          await this.fetchProducts(this.productIds)
+        } else {
+          showToast(translate("Order not Found"))
+        }
+      } catch (error) {
+        console.error(error)
+        showToast(translate("Something went wrong"))
+      }
+    },
+
+    async fetchProducts(productIds: any) {
+      const productIdFilter = productIds.reduce((filter: string, productId: any) => {
+        if (filter !== '') filter += ' OR '
+        return filter += productId;
+      }, '');
+
+      try {
+        const resp = await ProductService.fetchProducts({
+          "filters": ['productId: (' + productIdFilter + ')'],
+          "viewSize": process.env.VUE_APP_VIEW_SIZE
```

Comment:
View Size should be equal to the number of productIds for which data needs to be fetched.